network/netidentity.go

```diff
@@ -0,0 +1,202 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package network
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+// ProtocolConectionIdentityChallengeHeader is used to exchange IdentityChallenges
+const ProtocolConectionIdentityChallengeHeader = "X-Algorand-IdentityChallenge"
+
+// minimumProtocolVersion is used to evaluate if a peer's supported protocol
+// should include identityChallenge exchange
+var minimumProtocolVersion = [2]int64{2, 2}
```

Comment:
We'll be ticking up the protocol version for this, 2.3 at least if no other updates get in first